pkg/kubelet/cri/remote/remote_image.go

```diff
@@ -80,6 +141,39 @@ func (r *remoteImageService) ImageStatus(image *runtimeapi.ImageSpec) (*runtimea
 	ctx, cancel := getContextWithTimeout(r.timeout)
 	defer cancel()
 
+	// TODO: for refactoring common code blocks between the cri versions into
+	// one code block in the internal where possible examples:
+	// https://github.com/kubernetes/kubernetes/pull/104575/files#r705600987
+	// https://github.com/kubernetes/kubernetes/pull/104575/files#r696793706
+	if r.useV1API() {
+		return r.imageStatusV1(ctx, image)
+	}
+
+	return r.imageStatusV1alpha2(ctx, image)
+}
+
+func (r *remoteImageService) imageStatusV1alpha2(ctx context.Context, image *runtimeapi.ImageSpec) (*runtimeapi.Image, error) {
+	resp, err := r.imageClientV1alpha2.ImageStatus(ctx, &runtimeapiV1alpha2.ImageStatusRequest{
+		Image: v1alpha2ImageSpec(image),
+	})
+	if err != nil {
+		klog.ErrorS(err, "Get ImageStatus from image service failed", "image", image.Image)
+		return nil, err
+	}
+
+	if resp.Image != nil {
+		if resp.Image.Id == "" || resp.Image.Size_ == 0 {
```

Comment:
Size_(underscore) ?

Reply:
That's the name of the variable because protobuf generated that code. The code mainly originates from L186